Project/DAQ_System/Program/Src/Platform/I2C/STM/i2cHalWrapper_stmf4.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * i2cHalWrapper_stmf4.cpp
+ *
+ *  Created on: Mar 4, 2024
+ *      Author: Nico
+ */
+#include "i2cHalWrapper_stmf4.hpp"
+
+namespace platform {
+
+	i2cHalWrapperStmf4::i2cHalWrapperStmf4(I2C_HandleTypeDef &hi2c, uint8_t SAD)
+			: bx_i2c_(hi2c) {
+		slave_add_ = SAD;
+	}
+
+	i2cHalWrapperStmf4::~i2cHalWrapperStmf4(){}
+
+	void i2cHalWrapperStmf4::Master_Transmit(uint8_t *commands){
+		HAL_I2C_Master_Transmit(&bx_i2c_,slave_add_, commands, kMaxBytes, 0xFFFFFFFF);
+	}
+	void i2cHalWrapperStmf4::Master_Receive(uint8_t *response){
+		HAL_I2C_Master_Receive(&bx_i2c_, slave_add_, response, response_size_, 0xFFFFFFFF);
+	}
+	void i2cHalWrapperStmf4::SetResponse(uint8_t size){
+		response_size_ = size;
+	}
```

Comment:
If this configuration is expected of the library consumer to do before receiving/transmitting, I think it should be an argument of `Transmit()` and `Receive()`.

Unless, most drivers/libraries do it this way... 🤔💭

Thoughts?

Reply:
I also agree with Cristian's **Head Cannon** 🙉